### PR TITLE
[Batch] Add new data plane verison 2019-06-01

### PR DESCRIPTION
**Pull request by xingwu1**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by xingwu1** [comment]
@matthchr Please sign it off.

**Comment by xingwu1** [comment]
@sarangan12 I would like to ask for exemption for following error:

lintdiff: 
```
'Property named: \'nodeAgentSKUId\', for definition: \'ImageInformation\' must follow camelCase style. Example: \'nodeAgentSkuId\'.',
```

We would like to continue to use nodeAgentSKUId since we had old property has same name for same meaning. 

cSpell
```
specification/batch/data-plane/Microsoft.Batch/stable/2019-06-01.9.0/BatchService.json:433:7 - Unknown word (supportedimages)
```
This word is for url path.

**Comment by matthchr** [comment]
This looks good to me

**Comment by xingwu1** [comment]
@sarangan12  Hi, Could you review the PR? It has been open for 1 week.

**Comment by sarangan12** [comment]
@xingwu1 Please fix the error reported in https://dev.azure.com/azure-sdk/public/_build/results?buildId=38574&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d

**Comment by matthchr** [comment]
@xingwu1 - can you make the same change with is made in this PR (and/or rebase to master if the PR gets merged): https://github.com/Azure/azure-rest-api-specs/pull/6179/files

**Comment by xingwu1** [comment]
> @xingwu1 Please fix the error reported in https://dev.azure.com/azure-sdk/public/_build/results?buildId=38574&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d

@sarangan12 if you read my previous comment for exemption for this error.

We had used the exactly same property for other important object (https://github.com/Azure/azure-rest-api-specs/blob/f58c2f349207bda6f5ca4c010dd7f56c38ddb70e/specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json#L13056), and want to compact with the this property, we don't want to change the case for this property.

File: specification/batch/data-plane/Microsoft.Batch/stable/2019-06-01.9.0/BatchService.json [hunk 1]
```diff
@@ -0,0 +1,15947 @@
...
+            "required": true,
+            "type": "string",
+            "description": "The ID of the task whose file you want to delete."
+          },
+          {
+            "name": "filePath",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The path to the task file or directory that you
```
Comment (matthchr) [inline specification/batch/data-plane/Microsoft.Batch/stable/2019-06-01.9.0/BatchService.json]: Fix this comment by removing the `A maximum of 10 references may be specified on a given pool.` at the end of this comment

Comment (matthchr) [inline specification/batch/data-plane/Microsoft.Batch/stable/2019-06-01.9.0/BatchService.json]: `The login mode for the user` (I think this got missed in the merge)

**Review by matthchr — APPROVED** [review]

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**